src/devices/Display/Large4Digit14SegmentDisplay.cs

```diff
@@ -0,0 +1,326 @@
+        /// </summary>
+        /// <param name="value">integer value</param>
+        /// <param name="alignment">alignment on display (left or right, right is default)</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> must be between -999..9999</exception>
+        public void Write(int value, Alignment alignment = Alignment.Right)
+        {
+            if (value > 9999 || value < -999)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(value)} must be between -999..9999");
+            }
+
+            Write(value.ToString(), alignment);
+        }
+
+        /// <summary>
+        /// Write string value to display
+        /// </summary>
+        /// <param name="value">value to display, max 4 characters, or 5 characters if the 3rd character is ':' (this also turns on the center colon), or 6 characters if 1st character is also ':'</param>
+        /// <remarks>
+        /// * Unsupported characters will be replaced as whitespace
+        /// * This method clears the buffer before writing, so dots have to be reset afterwards
+        /// </remarks>
+        /// <param name="alignment">alignment on display (left or right, right is default)</param>
+        /// <exception cref="ArgumentException"><paramref name="value"/>[2] must be a ':'</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> can contain maximum 5 characters</exception>
+        public void Write(string value, Alignment alignment = Alignment.Left)
+        {
+            Clear();
+
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return;
+            }
+
+            switch (value.Length)
+            {
+                case 1 when alignment == Alignment.Right:
+                case 2 when alignment == Alignment.Right:
+                case 3 when alignment == Alignment.Right:
+                    value = value.PadLeft(MaxNumberOfDigits);
+                    break;
+                case 1 when alignment == Alignment.Left:
+                case 2 when alignment == Alignment.Left:
+                case 3 when alignment == Alignment.Left:
+                case 4:
+                case 5:
+                    // Nothing to do
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(value)} can contain maximum 5 characters");
+            }
+
+            Write(FontHelper14.GetString(value));
+        }
+
+        /// <summary>
+        /// Write an array of up to 4 bytes as hex
+        /// </summary>
+        /// <param name="values">Array of bytes</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="values"/>Length of array must be greater 0 and less than  MaxNumberOfDigits</exception>
+        public void WriteHex(byte[] values)
+        {
+            if (values.Length > 4)
```

Comment:
As you've define a constant for that, rather than 4, you should use this constant here.

Reply:
Done